### DISCUSSION TITLE
Queries

### DISCUSSION THREAD

**Discussion by MRizakis** [cjbarrie/academictwitteR#266] [Q&A]
Good evening,
I would first like to thank you and Justin for creating the package. I have a question on queries. I wanted to collect tweets from Glasgow, regarding the Covid-19 vaccine. 
I built a query and its this: 

```{r}
covid<-build_query("vaccine", is_retweet = FALSE, is_reply = FALSE, country = "GB", place = "glasgow", point_radius = c(55.858700, -4.295905, 20),lang= "en")
```
However, I though that when I used get_all_tweets(), I would just have to put the name of the query ("covid") and it would work. But it only returned tweets from various locations. Is there any guide to include the query in the get_all-tweets line of code I wrote. I read your documentation and it states it may be called separately and the output saved as a character object string to be input as a query parameter. However, due to my inexperience I do not know how to do it. Could you perhaps point me in the right direction? I apologise if this question is too basic but I am new to r programming. I thank you  in advance for your help.

Best regards,
M

**Comment by cjbarrie** [comment]
> Good evening,
> 
> I would first like to thank you and Justin for creating the package. I have a question on queries. I wanted to collect tweets from Glasgow, regarding the Covid-19 vaccine. 
> 
> I built a query and its this: 
> 
> 
> 
> ```{r}
> 
> covid<-build_query("vaccine", is_retweet = FALSE, is_reply = FALSE, country = "GB", place = "glasgow", point_radius = c(55.858700, -4.295905, 20),lang= "en")
> 
> ```
> 
> However, I though that when I used get_all_tweets(), I would just have to put the name of the query ("covid") and it would work. But it only returned tweets from various locations. Is there any guide to include the query in the get_all-tweets line of code I wrote. I read your documentation and it states it may be called separately and the output saved as a character object string to be input as a query parameter. However, due to my inexperience I do not know how to do it. Could you perhaps point me in the right direction? I apologise if this question is too basic but I am new to r programming. I thank you  in advance for your help.
> 
> 
> 
> Best regards,
> 
> M

Hard for us to say without showing what the `get_all_tweets()` query was that you used

My guess is that you specified "covid" as the query rather than `covid`

**Reply by MRizakis** [reply]
Good evening,
My apologies for the mistake. Here is another thing that I tried. First the query:

```{r}
covid<-build_query(query = "#COVIDIOTS", is_retweet = FALSE, 
is_reply = FALSE, country = "GB", 
place = "Glasgow", 
point_radius = c(55.860916, -4.251433, 25), lang= "en")
```

Now I called the query, and I incorporated it in the get_all_tweets() command:

```{r}
tweets_vac<-get_all_tweets(query = "#COVIDIOTS -is:retweet -is:reply place:Glasgow place_country:GB point_radius:[55.860916 -4.251433 25mi] lang:en", start_tweets = "2021-12-15T00:00:00Z", end_tweets = "2021-12-19T00:00:00Z" , bearer_token = bearer token, data_path = "/data",bind_tweets= FALSE, n= 5000)
```

However, it only returned 13 tweets. I asked another friend of mine and he told me that gathering tweets from a geographic location is de facto broken since most people do not activate the geotagging option. I speculate that this is causing the problem. I thank you in advance for your help.

Best regards,
M
